tests/test_cableBuffer.py

```diff
@@ -0,0 +1,134 @@
+        for line in profile:
+            match = re.search("(\d+)\s+(\w+)\s+(\d+)\s+(\d+)\s+(\d+)\s+(\d+)", line)
+            if match:
+                length = match.group(2)
+                profile_details[length][match.group(1)]['size'] = match.group(3)
+                profile_details[length][match.group(1)]['xon'] = match.group(4)
+                profile_details[length][match.group(1)]['xoff'] = match.group(5)
+                profile_details[length][match.group(1)]['threshold'] = match.group(6)
+        return profile_details
+
+    def test_CableBufferProfile(self, dvs):
+        cdb = swsscommon.DBConnector(4, dvs.redis_sock, 0)
+        adb = swsscommon.DBConnector(1, dvs.redis_sock, 0)
+        cfg_port_table = swsscommon.Table(cdb, "PORT")
+        cfg_cable_length_table = swsscommon.Table(cdb, "CABLE_LENGTH")
+        cfg_buffer_profile_table = swsscommon.Table(cdb, "BUFFER_PROFILE")
+        cfg_buffer_pg_table = swsscommon.Table(cdb, "BUFFER_PG")
+        asic_port_table = swsscommon.Table(adb, "ASIC_STATE:SAI_OBJECT_TYPE_PORT")
+        asic_profile_table = swsscommon.Table(adb, "ASIC_STATE:SAI_OBJECT_TYPE_BUFFER_PROFILE")
+
+        profile_details = self.getProfileDetails()
+        buffer_profiles = cfg_buffer_profile_table.getKeys()
+        expected_buffer_profiles_num = len(buffer_profiles)
+        # buffers.json used for the test defines 7 static profiles:
+        #    "ingress_lossless_profile"
+        #    "ingress_lossy_profile"
+        #    "egress_lossless_profile"
+        #    "egress_lossy_profile"
+        #    "pg_lossy_profile"
+        #    "q_lossless_profile"
+        #    "q_lossy_profile"
+        # check if they get the DB
+        assert expected_buffer_profiles_num == 7
+        assert len(asic_profile_table.getKeys()) == expected_buffer_profiles_num
+
+        for cable in self.cable_length_list:
+            for i in range(0, self.num_ports):
+                fvc = swsscommon.FieldValuePairs([("Ethernet%d" % (i*4), cable)])
+                cfg_cable_length_table.set("AZURE", fvc)
+            interface_cable_records = cfg_cable_length_table.getKeys()
+            (status, fvs) = cfg_cable_length_table.get('AZURE')
+            assert status == True
+            # check cable length is set for the interfaces
+            for interface in fvs:
+                assert interface[1] == cable
+
+            for speed in self.speed_list:
+                fvs = swsscommon.FieldValuePairs([("speed", speed)])
+                # set same speed on all ports
+                for i in range(0, self.num_ports):
+                    cfg_port_table.set("Ethernet%d" % (i*4), fvs)
+                time.sleep(1)
+
+                # check the speed was set
+                asic_port_records = asic_port_table.getKeys()
+                assert len(asic_port_records) == (self.num_ports + 1)  # +CPU port
+                num_set = 0
+                for k in asic_port_records:
```

Comment:
Is the CPU port assumed to be at the end of records? Else I think the below logic would fail right?

Reply:
There is an  conditional check for getting the speed Fv for all the port records .For cpu port case,  conditional check will get skipped and it will not  evaluated. So this has been handled in the below code.

                    (status, fvs) = asic_port_table.get(k)
                    assert status == True
                    for fv in fvs:
                        if fv[0] == "SAI_PORT_ATTR_SPEED":
                            assert fv[1] == speed
                            num_set += 1